### PR TITLE
Removed trav (and some other test/CI fixes)

### PR DESCRIPTION
**Pull request by Chillee**
In addition to replacing `trav` with for each loops, I also fixed the CI so that it would be considered "failing" if the compilation failed as well. I also changed `polygonUnionArea` to compare against 2 other implementations instead of a brute force.

The files without stress test coverage are: `IntPerm.h`, `3dHull.h`, `BellmanFord.h`,`BiconnectedComponents.h`, `dfsmatching.h`, `GeneralMatching.h`, `TopoSort.h`, `FastFourierTransform.h`, `FastSubsetTransform.h`, `AhoCorasick.h`.

I've inspected the changes I've made fairly closely and I'm reasonably certain they're correct.

### CODE REVIEW

File: content/combinatorial/IntPerm.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 int permToInt(vi& v) {
 	int use = 0, i = 0, r = 0;
-	trav(x, v) r = r * ++i + __builtin_popcount(use & -(1 << x)),
+	for(auto x:v) r = r * ++i + __builtin_popcount(use&-(1<<x)),
```
Comment (simonlindholm) [inline content/combinatorial/IntPerm.h:13]: int to save a character which can be spent on whitespace.

auto is good when the type could be replaced by something else by the user, but in this case only int fits. (I made the same comment in a few other places, but didn't go through all files)

File: content/data-structures/FenwickTree2d.h [hunk 2]
```diff
@@ -19,7 +19,7 @@ struct FT2 {
 		for (; x < sz(ys); x |= x + 1) ys[x].push_back(y);
 	}
 	void init() {
-		trav(v, ys) sort(all(v)), ft.emplace_back(sz(v));
+		for(auto &v: ys) sort(all(v)), ft.emplace_back(sz(v));
```
Comment (simonlindholm) [inline content/data-structures/FenwickTree2d.h:22]: `for (auto& v : ys)` to stay consistent; same elsewhere

File: content/geometry/ConvexHull.h [hunk 3]
```diff
@@ -13,8 +13,8 @@ Points on the edge of the hull between two other points are not considered part
 \includegraphics[width=\textwidth]{content/geometry/ConvexHull}
 \vspace{-6mm}
 \end{minipage}
- * Status: tested with Kattis problems convexhull
- * Time: O(n \log n)
+ * Status: stress-tested, tested with Kattis problems convexhull
+ * Time: O(n \log n),
```
Comment (simonlindholm) [inline content/geometry/ConvexHull.h:17]: weird comma

File: content/graph/EulerWalk.h [hunk 4]
```diff
@@ -26,6 +26,6 @@ vi eulerWalk(vector<vector<pii>>& gr, int nedges, int src=0) {
 			D[x]--, D[y]++;
 			eu[e] = 1; s.push_back(y);
 		}}
-	trav(x, D) if (x < 0 || sz(ret) != nedges+1) return {};
+	for(auto x: D) if (x < 0 || sz(ret) != nedges+1) return {};
```
Comment (simonlindholm) [inline content/graph/EulerWalk.h:29]: int

File: content/graph/DFSMatching.h [hunk 5]
```diff
@@ -28,7 +28,7 @@ int dfsMatching(vector<vi>& g, vi& btoa) {
 	vi vis;
 	rep(i,0,sz(g)) {
 		vis.assign(sz(btoa), 0);
-		trav(j,g[i])
+		for(auto j: g[i])
```
Comment (simonlindholm) [inline content/graph/DFSMatching.h:31]: int

File: content/graph/DFSMatching.h [hunk 6]
```diff
@@ -17,7 +17,7 @@
 bool find(int j, vector<vi>& g, vi& btoa, vi& vis) {
 	if (btoa[j] == -1) return 1;
 	vis[j] = 1; int di = btoa[j];
-	trav(e, g[di])
+	for(auto e: g[di])
```
Comment (simonlindholm) [inline content/graph/DFSMatching.h:20]: int

File: content/graph/BiconnectedComponents.h [hunk 7]
```diff
@@ -25,7 +25,7 @@ int Time;
 template<class F>
 int dfs(int at, int par, F& f) {
 	int me = num[at] = ++Time, e, y, top = me;
-	trav(pa, ed[at]) if (pa.second != par) {
+	for(auto pa: ed[at]) if (pa.second != par) {
```
Comment (simonlindholm) [inline content/graph/BiconnectedComponents.h:28]: structured bindings?

File: content/geometry/PolygonUnion.h [hunk 8]
```diff
@@ -37,7 +36,7 @@ double polyUnion(vector<vector<P>>& poly) {
 			}
 		}
 		sort(all(segs));
-		trav(s,segs) s.first = min(max(s.first, 0.0), 1.0);
+		for(auto &s: segs) s.first = min(max(s.first, 0.0), 1.0);
```
Comment (simonlindholm) [inline content/geometry/PolygonUnion.h:39]: good opportunity to use structured bindings here

File: doc/scripts/run-all.sh [hunk 9]
```diff
@@ -8,10 +8,12 @@ ulimit -s 524288 # For 2-sat test
 for test in $tests; do
     echo "$(basename $test): "
     start=`date +%s.%N`
-    g++ -std=c++14 -O2 $test && ./a.out
+    g++ -std=c++14 -O2 $test
+    retCode1=$?
+    ./a.out
```
Comment (simonlindholm) [inline doc/scripts/run-all.sh:13]: it doesn't make sense to run this if the compile failed. I thought &? would do the right thing with &&?

File: content/strings/Hashing.h [hunk 10]
```diff
@@ -51,4 +51,5 @@ vector<H> getHashes(string& str, int length) {
 	return ret;
 }
 
-H hashString(string& s) { H h{}; trav(c,s) h=h*C+c; return h; }
+
```
Comment (simonlindholm) [inline content/strings/Hashing.h:54]: remove whitespace

File: content/number-theory/MillerRabin.h [hunk 11]
```diff
@@ -15,7 +16,7 @@ bool isPrime(ull n) {
 	if (n < 2 || n % 6 % 4 != 1) return (n | 1) == 3;
 	ull A[] = {2, 325, 9375, 28178, 450775, 9780504, 1795265022},
 	    s = __builtin_ctzll(n-1), d = n >> s;
-	trav(a, A) {   // ^ count trailing zeroes
+	for(auto a: A) {   // ^ count trailing zeroes
```
Comment (simonlindholm) [inline content/number-theory/MillerRabin.h:19]: ull

File: content/graph/hopcroftKarp.h [hunk 12]
```diff
@@ -17,7 +17,7 @@
 bool dfs(int a, int L, vector<vi>& g, vi& btoa, vi& A, vi& B) {
 	if (A[a] != L) return 0;
 	A[a] = -1;
-	trav(b, g[a]) if (B[b] == L + 1) {
+	for(auto b: g[a]) if (B[b] == L + 1) {
```
Comment (simonlindholm) [inline content/graph/hopcroftKarp.h:20]: int, throughout this file

File: doc/scripts/run-all.sh (hunk 9 above)
Comment (Chillee) [reply doc/scripts/run-all.sh:13]: I was doing stupid stuff before.

**Comment by Chillee** [comment]
I'd prefer to move structured bindings to another PR - this PR is long enough as is.

File: content/graph/MaximumClique.h [hunk 13]
```diff
@@ -20,8 +20,8 @@ struct Maxclique {
 	vector<vi> C;
 	vi qmax, q, S, old;
 	void init(vv& r) {
-		trav(v,r) v.d = 0;
-		trav(v, r) trav(j, r) v.d += e[v.i][j.i];
+		for (auto& v : r) v.d = 0;
+		for (auto& v : r) for(auto j: r) v.d += e[v.i][j.i];
```
Comment (simonlindholm) [inline content/graph/MaximumClique.h:24]: nit: spacing

File: content/graph/TopoSort.h [hunk 14]
```diff
@@ -12,14 +12,14 @@
 
 vi topoSort(const vector<vi>& gr) {
 	vi indeg(sz(gr)), ret;
-	trav(li, gr) trav(x, li) indeg[x]++;
+	for (auto& li : gr) for(int x: li) indeg[x]++;
```
Comment (simonlindholm) [inline content/graph/TopoSort.h:15]: spacing

File: content/graph/hopcroftKarp.h [hunk 15]
```diff
@@ -33,13 +33,13 @@ int hopcroftKarp(vector<vi>& g, vi& btoa) {
...
 		cur.clear();
-		trav(a, btoa) if(a != -1) A[a] = -1;
+		for (int a : btoa) if(a != -1) A[a] = -1;
 		rep(a,0,sz(g)) if(A[a] == 0) cur.push_back(a);
 		/// Find all layers using bfs.
 		for (int lay = 1;; lay++) {
 			bool islast = 0;
 			next.clear();
-			trav(a, cur) trav(b, g[a]) {
+			for (int a : cur) for (int b: g[a]) {
```
Comment (simonlindholm) [inline content/graph/hopcroftKarp.h:42]: spacing

File: doc/example.h [hunk 16]
```diff
@@ -30,7 +30,7 @@ int exampleFunction1(vi mice, int* cats, int catsCount) {
 
 	rep(i, 1, catsCount) {
 		// This comment will be kept!
-		trav(it, mice) {
+		for(auto it: mice) {
```
Comment (simonlindholm) [inline doc/example.h:33]: spacing

**Pull request merged**